control-plane/grpc/src/operations/ha_node/client.rs

```diff
@@ -0,0 +1,53 @@
+use crate::{
+    context::{Client, TracedChannel},
+    operations::ha_node::traits::ClusterAgentOperations,
+};
+use common_lib::transport_api::{ReplyError, TimeoutOptions};
+use rpc::ha_cluster_agent::{ha_rpc_client::HaRpcClient, HaNodeInfo};
+use std::ops::Deref;
+use tonic::transport::Uri;
+
+/// Cluster-Agent RPC client
+pub struct ClusterAgentClient {
+    inner: Client<HaRpcClient<TracedChannel>>,
+}
+
+impl ClusterAgentClient {
+    /// creates a new base tonic endpoint with the timeout options and the address
+    pub async fn new<O: Into<Option<TimeoutOptions>>>(addr: Uri, opts: O) -> Self {
+        let client = Client::new(addr, opts, HaRpcClient::new).await;
+        Self { inner: client }
+    }
+}
+
+impl Deref for ClusterAgentClient {
+    type Target = Client<HaRpcClient<TracedChannel>>;
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+
+#[tonic::async_trait]
+impl ClusterAgentOperations for ClusterAgentClient {
+    #[tracing::instrument(
+        name = "ClusterAgentClient::register_node_agent",
+        level = "debug",
+        skip(self),
+        err
+    )]
+    async fn register_node_agent(
+        &self,
+        node_name: String,
+        end_point: String,
+    ) -> Result<(), ReplyError> {
+        let response = self
+            .client()
+            .register_node_agent(HaNodeInfo {
+                nodename: node_name,
+                endpoint: end_point,
+            })
+            .await?;
+        println!("registering node agent {:?}", response);
```

Comment:
Should be a trace rather than a println.
How often do we register, do we need to log this message everytime we register?

Reply:
At this point, the agent is already registered, so can we please change the text to something like "node agent successfully registered" or so. Thanks.